app/src/main/java/org/dhis2/utils/filters/OrgUnitFilterHolder.java

```diff
@@ -43,29 +48,37 @@ private void setUpAdapter() {
         OUFilterAdapter ouFilterAdapter = new OUFilterAdapter();
         localBinding.filterOrgUnit.ouRecycler.setAdapter(ouFilterAdapter);
 
+        localBinding.filterOrgUnit.orgUnitSearchEditText.setDropDownVerticalOffset(4);
         localBinding.filterOrgUnit.orgUnitSearchEditText.addTextChangedListener(new TextWatcher() {
             @Override
             public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
-
+                localBinding.filterOrgUnit.progress.setVisibility(View.VISIBLE);
             }
 
             @Override
             public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
                 if (charSequence.length() > 3) {
-                    currentOrgUnit = d2.organisationUnitModule().organisationUnits()
-                            .byDisplayName().like("%" + charSequence + "%").one().blockingGet();
-                    if (currentOrgUnit != null)
-                        localBinding.filterOrgUnit.orgUnitHint.setText(currentOrgUnit.displayName());
-                    else
-                        localBinding.filterOrgUnit.orgUnitHint.setText(null);
+                    List<OrganisationUnit> orgUnits = d2.organisationUnitModule().organisationUnits()
+                            .byDisplayName().like("%" + charSequence + "%")
+                            .blockingGet();
+                    currentOrgUnit = !orgUnits.isEmpty() ? orgUnits.get(0) : null;
+                    List<String> orgUnitsNames = new ArrayList<>();
+                    for (OrganisationUnit orgUnit: orgUnits) {
+                        orgUnitsNames.add(orgUnit.displayName());
+                    }
+                    if (!orgUnitsNames.isEmpty()) {
+                        ArrayAdapter<String> autoCompleteAdapter = new ArrayAdapter<>(itemView.getContext(), android.R.layout.simple_dropdown_item_1line, orgUnitsNames);
+                        localBinding.filterOrgUnit.orgUnitSearchEditText.setAdapter(autoCompleteAdapter);
+                        localBinding.filterOrgUnit.orgUnitSearchEditText.showDropDown();
+                    }
```

Comment:
Hello @mmmateos @Balcan, I know this was here before.

I was thinking about the agreement of viewmodels, but I'm not sure about using d2 in viewmodels or holder (coupling logic to UI component). Thoughs? Perhaps is a specific usecase